File: sofia-core/src/test/java/com/antwerkz/sofia/MessageFormatTest.java
package com.antwerkz.sofia;

import java.text.Format;
import java.text.MessageFormat;

import org.testng.Assert;
import org.testng.annotations.Test;

public class MessageFormatTest {
  @Test
  public void argCount() {
    count("Today''s date is {0,date,full} and now a number {1,number}");
  }

  private void count(final String pattern) {
    MessageFormat messageFormat = new MessageFormat(pattern);
    Format[] formats = messageFormat.getFormats();
    Assert.assertEquals(formats.length, 2);
  }
}

File: sofia-core/src/test/kotlin/com/antwerkz/sofia/MessageFormatTest.kt
package com.antwerkz.sofia

import java.text.MessageFormat
import org.testng.Assert
import org.testng.annotations.Test

class MessageFormatTest {
    @Test
    fun argCount() {
        count("Today''s date is {0,date,full} and now a number {1,number}")
    }

    private fun count(pattern: String) {
        val messageFormat = MessageFormat(pattern)
        val formats = messageFormat.getFormats()
        Assert.assertEquals(formats.size, 2)
    }
}
